include/cuco/detail/utils.hpp

```diff
@@ -49,5 +51,12 @@ auto get_grid_size(Kernel kernel, std::size_t block_size, std::size_t dynamic_sm
   return grid_size;
 }
 
+template <typename Iterator>
+constexpr inline int64_t distance(Iterator begin, Iterator end)
+{
+  // `int64_t` instead of arch-dependant `long int`
+  return static_cast<int64_t>(std::distance(begin, end));
```

Comment:
We could also add a `__device__` overload that uses `thrust::distance`. This way we could always call `cuco::distance` and don't have to worry about anything.

Reply:
> It may be a good idea to add a `static_assert` here that `Iterator` is a random access iterator

Done 

> We could also add a `__device__` overload that uses `thrust::distance`.

I've updated the algorithm in a way that `detail::distance` is only invoked on the host side.